src/digitalIdentity-backend/src/test/java/didentity/amos/digitalIdentity/Services/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,206 @@
+    }
+
+    @Test
+    void testAuthenticationWhenTokenEqualNull() {
+        assertFalse(authenticationService.authentication(null));
+        lastError = response401;
+        assertEquals(response401,lastError);
+    }
+
+    @Test
+    void testAuthenticationWhenTokenEqualAdmin() {
+        assertTrue(authenticationService.authentication("admin"));
+        assertNull(lastError);
+    }
+
+    @Test
+    void testAuthenticationWhenTokenEqualPassing() {
+        assertTrue(authenticationService.authentication("passing"));
+        assertNull(lastError);
+    }
+
+    @Test
+    void testAuthenticationWhenTokenEqualSomethingElse() {
+        assertFalse(authenticationService.authentication("John"));
+        lastError = response403;
+        assertEquals(response403,lastError);
+    }
+
+    @Test
+    void testGetError() {
+        assertNull((new AuthenticationService()).getError());
+    }
+
+    @Test
+    void testLoginWithSuccessfulLogin() {
+        user.setEmail("test@fau.de");
+        user.setId(1);
+        user.setName("Name0");
+        user.setPassword("test");
+        user.setSurname("Surname0");
+        user.setUserRole(UserRole.ADMIN);
+
+        Optional<User> ofResult = Optional.of(user);
+        when(userRepository.findByEmail(any())).thenReturn(ofResult);
+        ResponseEntity<String> actualLoginResult = authenticationService.login("test@fau.de", "test");
+
+        assertEquals("\"Login successful.\"", actualLoginResult.getBody());
+        assertEquals(200, actualLoginResult.getStatusCodeValue());
+        assertTrue(actualLoginResult.getHeaders().isEmpty());
+        verify(userRepository).findByEmail(any());
+    }
+
+    @Test
+    void testLoginWithWrongPassword() {
+        User user = mock(User.class);
+
+        when(user.getPassword()).thenReturn("SomePassword");
+        doNothing().when(user).setEmail(any());
+        doNothing().when(user).setId(any());
+        doNothing().when(user).setName(any());
+        doNothing().when(user).setPassword(any());
+        doNothing().when(user).setSurname(any());
+        doNothing().when(user).setUserRole(any());
+
+        user.setEmail("test@fau.de");
+        user.setId(1);
+        user.setName("Name0");
+        user.setPassword("test");
+        user.setSurname("Surname0");
+        user.setUserRole(UserRole.ADMIN);
+
+        Optional<User> ofResult = Optional.of(user);
+        when(this.userRepository.findByEmail(any())).thenReturn(ofResult);
+        ResponseEntity<String> actualLoginResult = this.authenticationService.login("test@fau.de", "test");
+
+        assertEquals("\"Password and username do not match.\"", actualLoginResult.getBody());
+        assertEquals(200, actualLoginResult.getStatusCodeValue());
+        assertTrue(actualLoginResult.getHeaders().isEmpty());
+        verify(this.userRepository).findByEmail(any());
+        verify(user).getPassword();
+        verify(user).setEmail(any());
```

Comment:
Why do you verify the function calls here but not for testLoginWithSuccessfulLogin()?

Reply:
it was to know if we really call the mock method because not everything happens here the way we want it to, but in the other case everything is supposed to go well

Reply:
Yes but I don't know why to ensure that the methods are called when they are not doing anything. So for the first two of course, but for the setter functions I just don't see it ^^

Reply:
I will make it clearly